keras/backend/theano_backend.py

```diff
@@ -324,8 +325,11 @@ def normalize_batch_in_training(x, gamma, beta,
 def batch_normalization(x, mean, std, beta, gamma, epsilon=0.0001):
     '''Apply batch normalization on x given mean, std, beta and gamma.
     '''
-    normed = T.nnet.bn.batch_normalization(x, gamma, beta, mean, std + epsilon,
-                                           mode='high_mem')
+    if theano.config.device.startswith('gpu'):
+        normed = dnn_batch_normalization_test(x, gamma, beta, mean, std ** 2, epsilon)
```

Comment:
This op takes variance as argument (http://deeplearning.net/software/theano_versions/dev/library/sandbox/cuda/dnn.html#batch-normalization) and this method takes std as argument, that's why **2. I didn't at mode properly. I should have actually. 

batch normalization implementation in theano (deeplearning.net/software/theano/library/tensor/nnet/bn.html) is per-activation, right?